daemon/graphdriver/overlay/overlay.go

```diff
@@ -23,26 +23,29 @@ import (
 // implementation of ApplyDiff()
 
 var (
+	// ErrApplyDiffFallback is returned to indicate that a normal ApplyDiff is applied as a fallback from Naive diff writer.
 	ErrApplyDiffFallback = fmt.Errorf("Fall back to normal ApplyDiff")
 )
 
-type ApplyDiffProtoDriver interface {
+type applyDiffProtoDriver interface {
 	graphdriver.ProtoDriver
 	ApplyDiff(id, parent string, diff archive.Reader) (size int64, err error)
 }
 
 type naiveDiffDriverWithApply struct {
 	graphdriver.Driver
-	applyDiff ApplyDiffProtoDriver
+	applyDiff applyDiffProtoDriver
 }
 
-func NaiveDiffDriverWithApply(driver ApplyDiffProtoDriver) graphdriver.Driver {
+// NaiveDiffDriverWithApply returns a NaiveDiff driver with custom ApplyDiff.
+func NaiveDiffDriverWithApply(driver applyDiffProtoDriver) graphdriver.Driver {
```

Comment:
I agree, good to spec out the interface implementation. Please review the comments. @tiborvass